src/backend/eventEmitters/loggerEmitter.ts

```diff
@@ -0,0 +1,13 @@
+import { BudgetTrackingEventEmitter, EventPublisher } from '@/backend/eventEmitters/EventEmitter';
+
+interface Logger {
+  info: (...params: any[]) => void
+}
+
+export function buildLoggerEmitter(logger: Logger): EventPublisher {
+  const loggerEmitter = new BudgetTrackingEventEmitter();
+  loggerEmitter.onAny((eventName, eventData) => {
+    logger.info(`${eventName}:`, eventData);
+  });
+  return loggerEmitter;
+}
```

Comment:
This is not a very smart method, but it is here for a reason.

Here, in the `eventEmitters` folder, in the context of the `eventEmitters`, we know the event types, so we can parse them to string (with special care for each event), and direct them to the right level.

So, continuously, we can use this method to

1. Print specific message depends on the Event.
2. Print to the console or to the file, by level (because the `logger` argument supports both `console.[level]` and `logger.[level]`